### ISSUE TITLE
U2X mode?

### ISSUE THREAD

**Issue by SpenceKonde** [SpenceKonde/megaTinyCore#188]
We could double the top speed of the UART available if we used U2X mode when it was appropriate - but when I tried to implement it, the code blew up by over 100 bytes - at which point i decided against it... 

```
  baud_setting = ((16 * F_CPU) / baud);
  // Disable CLK2X

  _written = false;



  //See #131 for more info on this
  #if !defined(USE_EXTERNAL_OSCILLATOR)
    #if (F_CPU==20000000UL || F_CPU==10000000UL || F_CPU==5000000UL) //this means we are on the 20MHz oscillator
      #ifdef UARTBAUD3V
        int8_t sigrow_val = SIGROW.OSC20ERR3V;
      #else
        int8_t sigrow_val = SIGROW.OSC20ERR5V;
      #endif
    #else //we are on 16MHz one
      #ifdef UARTBAUD3V
        int8_t sigrow_val = SIGROW.OSC16ERR3V;
      #else
        int8_t sigrow_val = SIGROW.OSC16ERR5V;
      #endif
    #endif
  #else
    int8_t sigrow_val = 0;
  #endif
  //baud_setting += (baud_setting * sigrow_val) / 1024;
  baud_setting *= (1024 + sigrow_val);
  baud_setting /= 1024;
  if ( baud_setting <255 ) { //used 255 in hopes it would be optimized heavily (aince )
    (*_hwserial_module).CTRLB |= USART_RXMODE_CLK2X_gc;
    //(*_hwserial_module).CTRLB &= ~USART_RXMODE_NORMAL_gc;
  } else {
    baud_setting=baud_setting>>1;
    (*_hwserial_module).CTRLB &= ~USART_RXMODE_CLK2X_gc;
    //(*_hwserial_module).CTRLB |= USART_RXMODE_NORMAL_gc;
  }

  // assign the baud_setting, a.k.a. BAUD (USART Baud Rate Register)
  (*_hwserial_module).BAUD = (uint16_t)((baud_setting + 1) / 2);

```

**Comment by SpenceKonde** [comment]
Main benefit would be that we could use 115200 baud at 1 MHz and 2 mbaud at 16 and 20 (barely, in the case of 16)

**Comment by SpenceKonde** [comment]
Implemented now in 2.1.4
